src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,54 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(String contact, boolean isPrivate, 
+    		String MESSAGE_CONTACT_CONSTRAINTS , String iden)throws IllegalValueException{
+    	this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        if(iden.equals("idenPhone")){
+        	if (!Phone.isValidPhone(trimmedContact)) {
+                throw new IllegalValueException(MESSAGE_CONTACT_CONSTRAINTS);
+            }
+        }else if(iden.equals("idenAddress")){
+        	if (!Address.isValidAddress(trimmedContact)) {
+                throw new IllegalValueException(MESSAGE_CONTACT_CONSTRAINTS);
+            }
+        }else if(iden.equals("idenEmail")){
+        	if (!Email.isValidEmail(trimmedContact)) {
+                throw new IllegalValueException(MESSAGE_CONTACT_CONSTRAINTS);
+            }
+        }
+        
+        this.value = trimmedContact;
+    }
+    
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Note that this comparison may be equal if email and address class have the same value for instance.